File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        i=0
        result=[]
        if(len(nums)==0):
            return []
        while(i<len(nums)):
            j=i+1
            while(j<len(nums)):
                if(nums[j-1]+1==nums[j]):
                    j+=1
                else:
                    break
            if(j-i==1):
                result.append(str(nums[i]))
            else:
                result.append(str(nums[i])+"->"+str(nums[j-1]))
            i=j    
        return result   
            
                    
                
            
        
    

        

File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> list=new ArrayList<>();
        int n = nums.length;
        if(n==0)
            return list;
        for(int i=0;i<n;i++){
            int j=i;
            while(j+1<n && nums[j]+1==nums[j+1]){
                j++;
            }
            if(i<j){
                list.add(nums[i]+"->"+nums[j]);
            }
            else{
                list.add(nums[i]+"");
            }
            i=j;
        }
        return list;
        
    }
}